app/Http/Controllers/Staff/PollController.php

```diff
@@ -99,4 +99,109 @@ public function store(StorePoll $request)
         return redirect()->route('staff.polls.index')
             ->withSuccess('Your poll has been created.');
     }
+
+    /**
+     * Poll Edit Form.
+     *
+     * @param $id
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function edit($id)
+    {
+        $poll = Poll::findOrFail($id);
+
+        return view('Staff.poll.edit', ['poll' => $poll]);
+    }
+
+    /**
+     * Update A New Poll.
+     *
+     * @param StorePoll $request
+     *
+     * @return Illuminate\Http\RedirectResponse
+     */
+    public function update(StorePoll $request, $id)
+    {
+        $poll = Poll::findOrFail($id);
+
+        $poll->title = $request->input('title');
+
+        if ($request->input('ip_checking')) {
+            $poll->ip_checking = true;
+        } else {
+            $poll->ip_checking = false;
+        }
+
+        if ($request->input('multiple_choice')) {
+            $poll->multiple_choice = true;
+        } else {
+            $poll->multiple_choice = false;
+        }
+
+        // Remove the deleted options in poll
+        $oldOptionIds = collect($poll->options)->map(function ($option) {
+            return $option->id;
+        })->all();
+
+        $existingOldOptionIds = collect($request->input('option-id'))->map(function ($id) {
+            return intval($id);
+        })->all();
+
+        $idsOfOptionToBeRemove = array_diff($oldOptionIds, $existingOldOptionIds);
+
+        foreach ($idsOfOptionToBeRemove as $id) {
+            $option = Option::findOrFail($id);
+            $option->delete();
+        }
+
+        // Update existing options
+        $existingOldOptionContents = collect($request->input('option-content'))->map(function ($content) {
+            return strval($content);
+        })->all();
+
+        if (count($existingOldOptionContents) === count($existingOldOptionIds)) {
+            $len = count($existingOldOptionContents);
+            for ($i = 0; $i < $len; $i++) {
+                $option = Option::findOrFail($existingOldOptionIds[$i]);
+                $option->name = $existingOldOptionContents[$i];
+                $option->save();
+            }
+        }
+
+        // Insert new options
+        $newOptions = collect($request->input('new-option-content'))->map(function ($content) {
+            return new Option(['name' => $content]);
+        });
+
+        $poll->options()->saveMany($newOptions);
+
+        // Last work from store()
+        $poll_url = hrefPoll($poll);
+
+        $this->chat->systemMessage(
+            "A poll has been updated [url={$poll_url}]{$poll->title}[/url] vote on it now! :slight_smile:"
+        );
+
+        $poll->save();
+
+        return redirect()->route('staff.polls.index')
+            ->withSuccess('Your poll has been edited.');
+    }
+
+    /**
+     * Delete A Poll.
+     *
+     * @param $id
+     *
+     * @return Illuminate\Http\RedirectResponse
+     */
+    public function destroy($id)
```

Comment:
Migrations need to be fixed as there is no delete cascade for poll options. There is only for poll votes. You could for now do these as we plan to refactor all migrations soon.
```
        $poll = Poll::with(['options'])->findOrFail($id);
        $poll->options->delete();
        $poll->delete();
```

Reply:
Should be done with https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/1126/commits/c6207c3f3b64acf5e08bce2e115e2d4c44b42f65. Reviews needed for sensitive db operation.